### ISSUE TITLE
Futures have inconsistent behavior in bash app fn body

### ISSUE THREAD

**Issue by yadudoc** [Parsl/parsl#35]
Eg : 

```python
@App('bash', dfk)
def app1(inputs=[], outputs=[], stdout=None, stderr=None, mock=False ):
    cmd_line = '''echo 'test' > {outputs[0]}'''

@App('bash', dfk)
def app2(inputs=[], outputs=[], stdout=None, stderr=None, mock=False ):

    with open('somefile.txt', 'w') as f:
        f.write("%s\n" % inputs[0])          #<--------------- Here inputs[0] is a DataFuture
    cmd_line = '''echo '{inputs[0]}' > {outputs[0]}'''
app1_future = app1(inputs = [],
                       outputs = [ "simple-out.txt"])
#app1_future.result()                                                                                                                                                    

app2_future = app2(inputs=[app1_future.outputs[0]],
                       outputs = ["simple-out2.txt"])
app2_future.result()
```
One fix is to evaluate the fn body entirely at app execution time. This need some work.

**Comment by yadudoc** [comment]
This issue is addressed by issue #37 and fixes are included in branch restructuring_libsubmit by 98f8d40f68479f032aa1a0bcbe20969d3370fc27

**Comment by yadudoc** [comment]
Tests passing on branch:restructuring_libsubmit. Closing.
